pkg/apiserver/registry/networkpolicy/clustergroupmember/rest.go

```diff
@@ -70,9 +81,52 @@ func (r *REST) Get(ctx context.Context, name string, options *metav1.GetOptions)
 		memberList.EffectiveMembers = effectiveMembers
 	}
 	memberList.Name = name
-	return memberList, nil
+	memberList.TotalMembers = int64(len(memberList.EffectiveMembers))
+	err = paginateMemberList(memberList, pageInfo)
+	return memberList, err
+}
+
+func (r *REST) NewGetOptions() (runtime.Object, bool, string) {
+	return &controlplane.PaginationGetOptions{}, false, ""
 }
 
 func (r *REST) NamespaceScoped() bool {
 	return false
 }
+
+// paginateMemberList returns paginated results if rational options are provided.
+// Paginated results are continuous only when there is no member change across multiple calls.
+// Returns full memberList if page number or page limit are given incorrectly.
+func paginateMemberList(memberList *controlplane.ClusterGroupMembers, pageInfo paginationInfo) error {
+	if memberList.TotalMembers == 0 {
+		memberList.TotalPages, memberList.CurrentPage = 0, 0
+		return nil
+	}
+	if pageInfo.pageLimit <= 0 {
+		memberList.TotalPages, memberList.CurrentPage = 1, 1
```

Comment:
Valid page starts from 1?

Reply:
Yes, valid page starts from 1. Evaluating here for `<=0` to handle error cases and when `pageLimit` was not provided, 0 was a default if I understand correctly.